src/main/kotlin/fi/metatavu/lipsanen/projects/milestones/tasks/TaskController.kt

```diff
@@ -228,6 +251,21 @@ class TaskController {
             milestone.endDate = newTask.endDate
         }
 
+        val dependentUser = if (newTask.dependentUserId != null) {
+            val user = userController.findUser(newTask.dependentUserId)
+                ?: throw UserNotFoundException(newTask.dependentUserId)
+
+            if (!projectController.hasAccessToProject(milestone.project, user.keycloakId)) {
+                logger.info("Assigning user ${user.id} to project ${milestone.project.id} because of the task assignment")
+                userController.assignUserToProjects(
+                    user = user,
+                    newProjects = listOf(milestone.project)
+                )
+            }
+
+            user
+        } else null
+
```

Comment:
```suggestion
        val dependentUser = newTask.dependentUserId?.let {
            val user = userController.findUser(newTask.dependentUserId)
                ?: throw UserNotFoundException(newTask.dependentUserId)

            if (!projectController.hasAccessToProject(milestone.project, user.keycloakId)) {
                logger.info("Assigning user ${user.id} to project ${milestone.project.id} because of the task assignment")
                userController.assignUserToProjects(
                    user = user,
                    newProjects = listOf(milestone.project)
                )
            }

            user
        }

```

Reply:
Fixed